src/document-info.js

```diff
@@ -56,10 +54,13 @@ export function documentInfoForDoc(nodeOrDoc) {
           canonicalUrl = parseUrl(canonicalTag.href).href;
         }
         const pageViewId = getPageViewId(ampdoc.win);
-        const res = {sourceUrl, canonicalUrl, pageViewId};
-        Object.defineProperty(res, 'sourceUrl', {
-          get: () => getSourceUrl(ampdoc.getUrl()),
-        });
+        const res = {
+          get sourceUrl() {
```

Comment:
This is still a getter property. Remove the `get` part, can it turns into a getter method. Maybe call it `getSourceUrl`.